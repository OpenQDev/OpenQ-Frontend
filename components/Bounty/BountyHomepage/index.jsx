// Third party
import React from 'react';
import { isOnlyContest } from '../../../services/utils/lib';

// Custom
import BountyList from '../../BountyList';
import UnexpectedErrorModal from '../../Utils/UnexpectedErrorModal';

const BountyHomepage = ({ watchedBounties, paginationObj, error, types, contractToggle }) => {
  const isContest = types && isOnlyContest(types);
  // Render
  return (
    <div>
      <div className='text-center bg-[#161B22] py-14 '>
        <div className='text-2xl font-bold'>{isContest ? 'Explore Hackathons' : 'Explore Issues'}</div>
        <div className='text-gray-500 text-md'>
          {isContest ? 'Hackathon bounties backed by OpenQ escrows' : 'GitHub issues backed by OpenQ escrows.'}
        </div>
      </div>

      {error ? (
        <UnexpectedErrorModal error={error} />
      ) : (
        <div className='lg:grid lg:grid-cols-extra-wide mx-4 sm:mx-8 xl:grid-cols-wide justify-center md:pr-3 pt-10'>
          <BountyList
            watchedBounties={watchedBounties}
            addCarousel={true}
            paginationObj={paginationObj}
            types={types}
            contractToggle={contractToggle}
          />
        </div>
      )}
    </div>
  );
};

export default BountyHomepage;
