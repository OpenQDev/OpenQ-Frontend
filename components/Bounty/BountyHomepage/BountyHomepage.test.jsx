import React from 'react';

import BountyHomepage from '.';
import Constants from '../../../test-utils/constant';
import ShallowRenderer from 'react-test-renderer/shallow';

describe('BountyHomepage', () => {
  const bounty = Constants.bounty;
  const categories = ['fixed price', 'split-price', 'contest'];

  it('should render UnexpectedErrorModal and match DOM Snapshot', () => {
    const shallow = new ShallowRenderer();
    shallow.render(<BountyHomepage bounty={bounty} address={bounty.bountyAddress} error={true} />);
    const tree = shallow.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });

  const test = (category) => {
    it('should render BountyDetails and match DOM Snapshot', () => {
      const shallow = new ShallowRenderer();
      shallow.render(<BountyHomepage bounty={bounty} address={bounty.bountyAddress} category={category} />);
      const tree = shallow.getRenderOutput();
      expect(tree).toMatchSnapshot(category);
    });
  };

  categories.forEach((category) => test(category));
});
