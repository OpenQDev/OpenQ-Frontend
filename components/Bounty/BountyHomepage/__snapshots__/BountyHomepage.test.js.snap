// vi Snapshot v1, https://goo.gl/fbAQLP

exports[`BountyHomepage should render BountyDetails and match DOM Snapshot: contest 1`] = `
<div>
  <div
    className="text-center bg-[#161B22] py-14 "
  >
    <div
      className="text-2xl font-bold"
    >
      Explore Issues
    </div>
    <div
      className="text-gray-500 text-md"
    >
      GitHub issues backed by OpenQ escrows.
    </div>
  </div>
  <div
    className="lg:grid lg:grid-cols-extra-wide mx-4 sm:mx-8 xl:grid-cols-wide justify-center md:pr-3 pt-10"
  >
    <BountyList
      addCarousel={true}
    />
  </div>
</div>
`;

exports[`BountyHomepage should render BountyDetails and match DOM Snapshot: fixed price 1`] = `
<div>
  <div
    className="text-center bg-[#161B22] py-14 "
  >
    <div
      className="text-2xl font-bold"
    >
      Explore Issues
    </div>
    <div
      className="text-gray-500 text-md"
    >
      GitHub issues backed by OpenQ escrows.
    </div>
  </div>
  <div
    className="lg:grid lg:grid-cols-extra-wide mx-4 sm:mx-8 xl:grid-cols-wide justify-center md:pr-3 pt-10"
  >
    <BountyList
      addCarousel={true}
    />
  </div>
</div>
`;

exports[`BountyHomepage should render BountyDetails and match DOM Snapshot: split-price 1`] = `
<div>
  <div
    className="text-center bg-[#161B22] py-14 "
  >
    <div
      className="text-2xl font-bold"
    >
      Explore Issues
    </div>
    <div
      className="text-gray-500 text-md"
    >
      GitHub issues backed by OpenQ escrows.
    </div>
  </div>
  <div
    className="lg:grid lg:grid-cols-extra-wide mx-4 sm:mx-8 xl:grid-cols-wide justify-center md:pr-3 pt-10"
  >
    <BountyList
      addCarousel={true}
    />
  </div>
</div>
`;

exports[`BountyHomepage should render UnexpectedErrorModal and match DOM Snapshot 1`] = `
<div>
  <div
    className="text-center bg-[#161B22] py-14 "
  >
    <div
      className="text-2xl font-bold"
    >
      Explore Issues
    </div>
    <div
      className="text-gray-500 text-md"
    >
      GitHub issues backed by OpenQ escrows.
    </div>
  </div>
  <UnexpectedErrorModal
    error={true}
  />
</div>
`;
